Update dependencies and update for Java 21

colin-lamed opened the pull request.

danlaunl commented:
Hi @colin-lamed , thanks for your work. I can't get the tests to run as they ran previously:
-manually by running sbt findSecBugs in the test-project 
-automated test which can be run by this command sbt scripted from the top dir

Could u please update, or update the readme?
ps this level of sbt is beyond me , but feel free to explain :) Am looking forward to merging this

colin-lamed commented:
Hi @danlauni what issue are you're having? What version of Java are you using?

I was running them with Java 11 - the built plugin will work with Java 11 or 21, but some of the scripted tests use sbt `1.3.3`.  They need to be updated to at least sbt `1.9.7` (and Scala `2.12.18`/`2.13.12`) to be able to run all the tests with Java 21.
We could either add something in the README, or even update the tests to use Java 21 compatible versions of Scala and sbt.

I am not aware of an issue with running `sbt findSecBugs` in the `test-project` with either Java version, since it's on compatible version of sbt `1.10.10` (which determines the default Scala version as `2.12.20` - but this could be set explicitly).

danlaunl commented:
@colin-lamed thx. I am running with Java 21, and due to your comments I vaguely see the need for something in the readme about java versions (and maybe even sbt and scala versions?) . I guess I would like to know how to run everything "out of the box" for java 21 and any relevant scala and sbt versions first (ie imagine if we didnt care about breaking changes; just git clone and run sbt scripted. )
ie. to Make a branch called java21 in addition.

And then write something small  in the readme to mention that branch (feel free to write this too!)

Yes, you're right `sbt findSecBugs` in the test-project runs fine. sorry!

---

 On updating all instances of "1.3.3" (sbt version) to 1.9.7 (feel free to push it to 1.10.10) one extra test succeeds
 > [info] + test-issues/test-issue-26
 
 But 3 still fail
 > [error] (scripted) Failed tests:
> [error] 	test-output/test-error-found
> [error] 	test-output/test-html-output
> [error] 	test-output/test-sbt-1.10.x

maybe something about compiler-bridge. see full logs here. i don't know where scala 2.12.3 is coming from
https://pastebin.com/XYCsANn0

my >scala --version returns 
Scala code runner version: 1.7.1
Scala version (default): 3.7.0

colin-lamed commented:
Hi @danlauni, I have made another push. I have updated all the tests to use sbt `1.9.7`, which is the first one to support Java `21`. All tests will now run with either Java `11` or `21`. The produced artefact will also run OK with older sbt/Scala, as long as they are using Java `11`. I think documenting general sbt/Scala/Java compatibility is out-of-scope for this repo's documentation.

Previously the tests were all run with sbt `1.1.0`, with a specific test to run with sbt `1.3.x` - I'm not sure what this was testing, but I have updated this one to sbt `1.11.x` (the current latest).
